### PR TITLE
wip: Hashmap accounts index perf

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_index.rs [hunk 1]
```diff
@@ -631,7 +633,9 @@ impl<'a, T> AccountsIndexIterator<'a, T> {
                 result.push((k, v));
             }
         }
-        result.sort_unstable_by(|a, b| a.0.cmp(b.0));
+        if !collect_all_unsorted {
```
Comment (jeffwashington) [inline runtime/src/accounts_index.rs:636]: @ryoqun, thoughts on this? We don't need to sort for rent collection.

File: runtime/src/accounts_index.rs [hunk 2]
```diff
@@ -707,10 +712,12 @@ impl<'a, T: 'static + Clone> Iterator for AccountsIndexIterator<'a, T> {
...
-            for (pubkey, account_map_entry) in
-                Self::range(&i.read().unwrap(), (self.start_bound, self.end_bound))
-            {
-                if chunk.len() >= ITER_BATCH_SIZE {
+            for (pubkey, account_map_entry) in Self::range(
+                &i.read().unwrap(),
+                (self.start_bound, self.end_bound),
+                self.collect_all_unsorted,
+            ) {
+                if chunk.len() >= ITER_BATCH_SIZE && !self.collect_all_unsorted {
```
Comment (jeffwashington) [inline runtime/src/accounts_index.rs:720]: @ryoqun, and for rent collection we don't need to limit the # of results we return at once.

Comment (ryoqun) [reply runtime/src/accounts_index.rs:720]: yes too. (ideally, we would like to parallelize the eager rent collection.. or change the whole rent collection mechanism altogether with @aeyakovenko 's account.deletion_slot based rent collection approch...)

Comment (jeffwashington) [reply runtime/src/accounts_index.rs:720]: @carllin @sakridge - this is a commit that gets rid of the sorting for rent collection to understand that more deeply. Initial bench-tps/perf-test numbers indicate this is better than current master. I don't have accurate numbers yet. There is a regression in master that behzad is looking at.

**Pull request closed without merging**